src/components/cax-companyData.tsx

```diff
@@ -158,19 +158,23 @@ export const CompanyDataCax = () => {
   }, [identifierType, identifierNumber, country])
 
   useEffect(() => {
-    setBpn(companyDetails?.bpn)
-    setLegalEntity(companyDetails?.name)
-    setRegisteredName(companyDetails?.name)
-    setStreetHouseNumber(companyDetails?.streetName)
-    setRegion(companyDetails?.region)
-    setPostalCode(companyDetails?.zipCode)
-    setCity(companyDetails?.city)
-    setCountry(companyDetails?.countryAlpha2Code)
-    setUniqueIds(companyDetails?.uniqueIds)
-    setIdentifierNumber(companyDetails?.uniqueIds?.[0]?.value)
-    setIdentifierType(companyDetails?.uniqueIds?.[0]?.type)
+    setFields(companyDetails);
   }, [companyDetails])
 
+  const setFields = (bpnDetails: any) =>{
+    setBpn(bpnDetails? bpnDetails?.bpn : '')
+    setLegalEntity(bpnDetails? bpnDetails?.name : '')
+    setRegisteredName(bpnDetails? bpnDetails?.name : '')
+    setStreetHouseNumber(bpnDetails? bpnDetails?.streetName : '')
+    setRegion(bpnDetails? bpnDetails?.region : '')
+    setPostalCode(bpnDetails? bpnDetails?.zipCode : '')
+    setCity(bpnDetails? bpnDetails?.city : '')
+    setCountry(bpnDetails? bpnDetails?.countryAlpha2Code : '')
+    setUniqueIds(bpnDetails? bpnDetails?.uniqueIds : '')
+    setIdentifierNumber(bpnDetails? bpnDetails?.uniqueIds?.[0]?.value : '')
+    setIdentifierType(bpnDetails? bpnDetails?.uniqueIds?.[0]?.type : '')
```

Comment:
Hey Jordan, 

I would try to stay DRY and make the code easier to read by either:
1. Using nullish coalescing:
```
setBpn(bnpDetails?.bpn ?? "")
```
2. Using object destructuring with default values
```
const { bpn = "", name = "" } = companyDetails
setBpn(bpn)
```
I find option 2 easier to read, but choose whatever works best for you.

Nice job to have created a single function!

Reply:
@ybidois thanks for the suggestion!

I decided to update to option 1, its also easy to read imo and requires less code.